cli/command/plugin/upgrade_test.go

```diff
@@ -0,0 +1,52 @@
+package plugin
+
+import (
+	"io"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/docker/cli/internal/test"
+	"github.com/docker/docker/api/types"
+	"github.com/pkg/errors"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/golden"
+)
+
+func TestUpgradePromptTermination(t *testing.T) {
+	cli := test.NewFakeCli(&fakeClient{
+		pluginUpgradeFunc: func(name string, options types.PluginInstallOptions) (io.ReadCloser, error) {
+			return nil, errors.New("should not be called")
+		},
+		pluginInspectFunc: func(name string) (*types.Plugin, []byte, error) {
+			return &types.Plugin{
+				ID:              "5724e2c8652da337ab2eedd19fc6fc0ec908e4bd907c7421bf6a8dfc70c4c078",
+				Name:            "foo/bar",
+				Enabled:         false,
+				PluginReference: "localhost:5000/foo/bar:latest",
+			}, []byte{}, nil
+		},
+	})
+	cmd := newUpgradeCommand(cli)
+	// need to set a remote address that does not match the plugin reference sent by the `pluginInspectFunc`
+	cmd.SetArgs([]string{"plugin", "localhost:5000/foo/bar:v1.0.0"})
+
+	errChan := make(chan error)
+	go func() {
+		errChan <- cmd.Execute()
+	}()
+
+	// wait for the prompt to be displayed
+	time.Sleep(100 * time.Millisecond)
+
+	syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+	// sigint and sigterm are caught by the prompt
+	// this allows us to gracefully exit the prompt with a 0 exit code
+	select {
+	case err := <-errChan:
+		assert.Error(t, err, "canceling upgrade request")
+	case <-time.After(100 * time.Millisecond):
+		t.Fatal("Plugin Upgrade did not return after SIGINT")
+	}
```

Comment:
Looks like this chunk of code is repeated in various tests. I haven't looked closely, but wondering if this would be a candidate for factoring out into an (`internal/`?) test-utility.

Reply:
Yes it's somewhat repetitive since most of the functions follow the same pattern. I will move this to minimize repetition

Reply:
It looks like we have an `internal/` package already, that also has some test-utilities (including, I see, a fakeCLI as well, so _perhaps_ even the fakeCLI parts could be integrated there, but I didn't look closely!); https://github.com/docker/cli/blob/master/internal/test/cli.go